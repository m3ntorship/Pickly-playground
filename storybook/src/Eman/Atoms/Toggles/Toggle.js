import React, { useState } from "react";
import PropTypes from "prop-types";

export const TOGGLE = {
  SIZE: {
    DEFAULT: "Default",
    SMALL: "Small",
  },
};

export const Toggle = ({ size, disabled }) => {
  const [isChecked, setIsChecked] = useState(true);

  const handleCheck = () => {
    setIsChecked(!isChecked);
  };

  const divSize = size === TOGGLE.SIZE.DEFAULT ? "w-10 h-6" : "w-7 h-4";
  const spanSize = size === TOGGLE.SIZE.DEFAULT ? "w-4 h-4" : "w-2.5 h-2.5";

  return (
    <>
      {isChecked && (
        <div
          className={`relative bg-primary-shd7 hover:bg-primary-shd6 focus:border-primary-shd5 focus:outline-none focus:border rounded-full flex items-center flex-row-reverse ${divSize} ${
            disabled && "bg-primary-shd7"
          }`}
        >
          <span
            className={`mr-1 inline-block rounded-lg bg-primary ${spanSize} ${
              disabled && "bg-primary-shd4"
            }`}
          ></span>
          <input
            disabled={disabled}
            onClick={handleCheck}
            type="checkbox"
            className="opacity-0 absolute cursor-pointer w-full h-full"
          />
        </div>
      )}
      {!isChecked && (
        <div
          className={`relative bg-grey-shd4 hover:bg-grey-shd3 focus:border-grey-shd4 focus:outline-none rounded-full flex items-center ${divSize} ${
            disabled && "bg-grey-shd6"
          }`}
        >
          <span
            className={`inline-block ml-1 rounded-lg bg-white ${spanSize} ${
              disabled && "bg-white"
            }`}
          ></span>
          <input
            disabled={disabled}
            onClick={handleCheck}
            type="checkbox"
            className="opacity-0 absolute cursor-pointer w-full h-full"
          />
        </div>
      )}
    </>
  );
};

Toggle.propTypes = {
  disabled: PropTypes.bool,
  size: PropTypes.oneOf(["Default", "Small"]),
};
