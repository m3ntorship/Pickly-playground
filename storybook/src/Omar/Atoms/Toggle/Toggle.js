import { React, useState } from "react";
import PropTypes from "prop-types";
export const TOGGLE = {
  SIZE: {
    SMALL: "small",
    DEFAULT: "default",
  },
};
export const Toggler = ({ disabled, size, label }) => {
  const [isChecked, setisChecked] = useState(false);
  const toggleSize = size === TOGGLE.SIZE.DEFAULT ? "w-10 h-6" : "w-8 h-4";
  const spanSize = size === TOGGLE.SIZE.DEFAULT ? "w-4 h-4" : "w-3 h-3";
  label = isChecked ? "ON" : "OFF";
  const handleCheck = () => {
    setisChecked(!isChecked);
  };
  return (
    <>
      <label htmlFor="labelId">{label}</label>
      {isChecked && (
        <div
          className={`relative ${toggleSize} bg-primary-shd7 hover:bg-primary-shd6 focus:outline-none focus:border focus:border-primary-shd5  rounded-full flex items-center flex-row-reverse ${
            disabled && "bg-primary-shd7 "
          }`}
        >
          <span
            className={` inline-block mr-1 rounded-lg bg-primary ${spanSize} ${
              disabled && "bg-primary-shd4"
            }
            `}
          ></span>
          <input
            id="labelId"
            type="checkbox"
            className={`opacity-0 absolute cursor-pointer w-full h-full ${
              disabled && "cursor-not-allowed"
            }`}
            disabled={disabled}
            onClick={handleCheck}
          />
        </div>
      )}
      {!isChecked && (
        <div
          className={`relative ${toggleSize} bg-grey-shd4 hover:bg-grey-shd3 focus:outline-none focus:border focus:border-grey-shd4  rounded-full flex items-center flex-row-reverse ${
            disabled && "bg-grey-shd6"
          }`}
        >
          <span
            className={`inline-block ${
              size === TOGGLE.SIZE.SMALL ? "mr-4" : "mr-5"
            } rounded-lg bg-white ${spanSize} ${disabled && "bg-white"}
            `}
          ></span>
          <input
            id="labelId"
            type="checkbox"
            className="opacity-0 absolute cursor-pointer w-full h-full"
            disabled={disabled}
            onClick={handleCheck}
          />
        </div>
      )}
    </>
  );
};

Toggler.propTypes = {
  disabled: PropTypes.bool,
  size: PropTypes.oneOf(["small", "default"]),
};
