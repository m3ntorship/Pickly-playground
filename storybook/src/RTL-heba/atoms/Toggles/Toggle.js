import React from "react";
import PropTypes from "prop-types";

export const TOGGLE = {
  STATUS: {
    ON: "on",
    OFF: "off",
  },
  SIZE: {
    DEFAULT: "Default",
    SMALL: "Small",
  },
};

export const Toggle = ({ status, size }) => {
  return (
    <div title="Toggle">
      {size === TOGGLE.SIZE.DEFAULT && status === TOGGLE.STATUS.ON && (
        <button className="bg-primary-shd7 hover:bg-primary-shd6 focus:border-primary-shd5 focus:outline-none focus:border w-10 h-6 rounded-full flex items-center flex-row-reverse">
          <span className="w-4 h-4 mr-1 inline-block rounded-lg bg-primary"></span>
        </button>
      )}
      {size === TOGGLE.SIZE.DEFAULT && status === TOGGLE.STATUS.OFF && (
        <button className="bg-grey-shd4 hover:bg-grey-shd3 focus:border-grey-shd4 focus:outline-none w-10 h-6 rounded-full flex items-center">
          <span className="w-4 h-4 inline-block ml-1 rounded-lg bg-white"></span>
        </button>
      )}
      {size === TOGGLE.SIZE.SMALL && status === TOGGLE.STATUS.ON && (
        <button className="bg-primary-shd7 hover:bg-primary-shd6 focus:border-primary-shd5 focus:outline-none focus:border w-7 h-4 rounded-full flex items-center flex-row-reverse">
          <span className="w-2.5 h-2.5 mr-1 inline-block rounded-lg bg-primary"></span>
        </button>
      )}
      {size === TOGGLE.SIZE.SMALL && status === TOGGLE.STATUS.OFF && (
        <button className="bg-grey-shd4 hover:bg-grey-shd3 focus:border-grey-shd4 focus:outline-none w-7 h-4 rounded-full flex items-center">
          <span className="w-2.5 h-2.5 inline-block ml-1 rounded-lg bg-white"></span>
        </button>
      )}
    </div>
  );
};

Toggle.propTypes = {
  status: PropTypes.oneOf(["on", "off"]),
  size: PropTypes.oneOf(["Default", "Small"]),
};
