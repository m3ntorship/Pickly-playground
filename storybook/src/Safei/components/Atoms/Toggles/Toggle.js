import React from "react";
import "./toggle.css";

const Toggle = (props) => {
  let size = "1";
  if (props.size === "sm") {
    size = "0.666";
  } else {
    size = "1";
  }

  return (
    <div style={{ transform: `scale(${size})` }}>
      <label className="switch">
        <input type="checkbox" disabled={props.disabled} checked={props.on} />
        <span className="slider round"></span>
      </label>
    </div>
  );
};
export default Toggle;
