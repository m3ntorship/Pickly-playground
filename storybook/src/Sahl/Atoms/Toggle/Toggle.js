import React, { useState } from "react";
import PropTypes from "prop-types";

export default function Toggle(props) {
  // Props
  const { disabled = false, size = "lg", onChange } = props;

  const [checked, setChecked] = useState(false);

  /////////////////////////////////////////
  // Style Classes
  ////////////////
  // container class
  let containerClasses = [
    "relative",
    "inline-block",
    "rounded-full",
    "focus:outline-none",
    "focus:border-2",
    "group",
  ];
  size === "lg"
    ? containerClasses.push("w-10 h-6 ")
    : containerClasses.push("w-7 h-4 ");

  if (disabled) containerClasses.push("opacity-30");

  checked
    ? containerClasses.push("focus:border-primary-shd1")
    : containerClasses.push("focus:border-grey");

  // Background Class
  let backgroundClasses = [
    "absolute",
    "cursor-pointer",
    "top-0",
    "left-0",
    "right-0",
    "bottom-0",
    "w-full",
    "h-full",
    "block",
    "transition",
    "duration-300",
    "rounded-full",
  ];
  checked
    ? (backgroundClasses = [
        ...backgroundClasses,
        "bg-primary-shd7",
        "group-hover:bg-primary-shd5",
        "group-focus:border",
        "group-focus:border-primary-shd3",
      ])
    : (backgroundClasses = [
        ...backgroundClasses,
        "bg-grey-shd4",
        "group-hover:bg-grey-shd3",
        "group-focus:border",
        "group-focus:border-solid",
        "group-focus:border-primary-shd1",
      ]);

  // Toggle Classes

  let toggleClasses = [
    "absolute",
    "left-1",
    "rounded-full",
    "transition",
    "duration-300",
  ];
  size === "lg"
    ? (toggleClasses = [...toggleClasses, "w-4 h-4", "bottom-1"])
    : (toggleClasses = [...toggleClasses, "w-3 h-3", "bottom-0.5"]);
  !checked
    ? toggleClasses.push("bg-white")
    : size === "lg"
    ? (toggleClasses = [
        ...toggleClasses,
        "bg-primary-shd8",
        "transform",
        "translate-x-4",
      ])
    : (toggleClasses = [
        ...toggleClasses,
        "bg-primary-shd8",
        "transform",
        "translate-x-2",
      ]);
  /////////////////////////////////////////
  // Handlers
  const onClick = () => {
    setChecked(!checked);
  };

  return (
    <label className={containerClasses.join(" ")}>
      <input
        disabled={disabled}
        type="checkbox"
        className="hidden"
        onChange={onChange}
        onClick={onClick}
      />
      {/* Background */}
      <span className={backgroundClasses.join(" ")}>
        {/* Toggle */}
        <span className={toggleClasses.join(" ")}></span>
      </span>
    </label>
  );
}

Toggle.propTypes = {
  disabled: PropTypes.bool,
  size: PropTypes.oneOf(["sm", "lg"]),
  // onChange: PropTypes.func,
};
