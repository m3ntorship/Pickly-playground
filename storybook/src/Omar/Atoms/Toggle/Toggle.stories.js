import React from "react";
import { Toggler, TOGGLE } from "./Toggle";
export default {
  title: "Omar/Atoms/Toggle",
  component: Toggler,
  size: {
    control: {
      type: "select",
      options: [TOGGLE.SIZE.DEFAULT, TOGGLE.SIZE.SMALL],
    },
  },
};

const Template = (args) => <Toggler {...args} />;
export const toggle = Template.bind({});
toggle.args = {
  disabled: false,
  size: TOGGLE.SIZE.DEFAULT,
};
