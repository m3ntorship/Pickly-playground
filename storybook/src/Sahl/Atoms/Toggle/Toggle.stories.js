import React from "react";

import Toggle from "./Toggle";

const defaultExport = {
  title: "Sahl/Atoms/Toggle",
  component: Toggle,
  argTypes: {
    onChange: { control: "checked" },
  },
};

export default defaultExport;

const Template = (args) => <Toggle {...args} />;

export const Enabled = Template.bind({});
Enabled.args = {
  disabled: false,
  size: "lg",
};

export const Disabled = Template.bind({});
Disabled.args = {
  disabled: true,
  size: "small",
};

export const Large = Template.bind({});
Large.args = {
  size: "large",
};

export const Small = Template.bind({});
Small.args = {
  size: "small",
};
