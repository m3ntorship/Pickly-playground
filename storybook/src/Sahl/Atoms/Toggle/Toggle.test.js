import React from "react";
import renderer from "react-test-renderer";
import Toggle from "./Toggle";

describe("Toggle Component", () => {
  it("should render with large OFF toggle as default without any props ", () => {
    const tree = renderer.create(<Toggle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render change toggle from OFF to ON and vice versa when clicking on it ", () => {
    const tree = renderer.create(<Toggle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render small OFF toggle when passing size=sm  ", () => {
    const tree = renderer.create(<Toggle size="sm" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render large OFF toggle when passing size=lg  ", () => {
    const tree = renderer.create(<Toggle size="sm" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render disable toggle when passing disabled ", () => {
    const tree = renderer.create(<Toggle disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
