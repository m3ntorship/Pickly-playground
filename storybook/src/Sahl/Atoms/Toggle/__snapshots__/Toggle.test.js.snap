// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Component should render change toggle from OFF to ON and vice versa when clicking on it  1`] = `
<label
  className="relative inline-block rounded-full focus:outline-none focus:border-2 group w-10 h-6  focus:border-grey"
>
  <input
    className="hidden"
    disabled={false}
    onClick={[Function]}
    type="checkbox"
  />
  <span
    className="absolute cursor-pointer top-0 left-0 right-0 bottom-0 w-full h-full block transition duration-300 rounded-full bg-grey-shd4 group-hover:bg-grey-shd3 group-focus:border group-focus:border-solid group-focus:border-primary-shd1"
  >
    <span
      className="absolute left-1 rounded-full transition duration-300 w-4 h-4 bottom-1 bg-white"
    />
  </span>
</label>
`;

exports[`Toggle Component should render disable toggle when passing disabled  1`] = `
<label
  className="relative inline-block rounded-full focus:outline-none focus:border-2 group w-10 h-6  opacity-30 focus:border-grey"
>
  <input
    className="hidden"
    disabled={true}
    onClick={[Function]}
    type="checkbox"
  />
  <span
    className="absolute cursor-pointer top-0 left-0 right-0 bottom-0 w-full h-full block transition duration-300 rounded-full bg-grey-shd4 group-hover:bg-grey-shd3 group-focus:border group-focus:border-solid group-focus:border-primary-shd1"
  >
    <span
      className="absolute left-1 rounded-full transition duration-300 w-4 h-4 bottom-1 bg-white"
    />
  </span>
</label>
`;

exports[`Toggle Component should render large OFF toggle when passing size=lg   1`] = `
<label
  className="relative inline-block rounded-full focus:outline-none focus:border-2 group w-7 h-4  focus:border-grey"
>
  <input
    className="hidden"
    disabled={false}
    onClick={[Function]}
    type="checkbox"
  />
  <span
    className="absolute cursor-pointer top-0 left-0 right-0 bottom-0 w-full h-full block transition duration-300 rounded-full bg-grey-shd4 group-hover:bg-grey-shd3 group-focus:border group-focus:border-solid group-focus:border-primary-shd1"
  >
    <span
      className="absolute left-1 rounded-full transition duration-300 w-3 h-3 bottom-0.5 bg-white"
    />
  </span>
</label>
`;

exports[`Toggle Component should render small OFF toggle when passing size=sm   1`] = `
<label
  className="relative inline-block rounded-full focus:outline-none focus:border-2 group w-7 h-4  focus:border-grey"
>
  <input
    className="hidden"
    disabled={false}
    onClick={[Function]}
    type="checkbox"
  />
  <span
    className="absolute cursor-pointer top-0 left-0 right-0 bottom-0 w-full h-full block transition duration-300 rounded-full bg-grey-shd4 group-hover:bg-grey-shd3 group-focus:border group-focus:border-solid group-focus:border-primary-shd1"
  >
    <span
      className="absolute left-1 rounded-full transition duration-300 w-3 h-3 bottom-0.5 bg-white"
    />
  </span>
</label>
`;

exports[`Toggle Component should render with large OFF toggle as default without any props  1`] = `
<label
  className="relative inline-block rounded-full focus:outline-none focus:border-2 group w-10 h-6  focus:border-grey"
>
  <input
    className="hidden"
    disabled={false}
    onClick={[Function]}
    type="checkbox"
  />
  <span
    className="absolute cursor-pointer top-0 left-0 right-0 bottom-0 w-full h-full block transition duration-300 rounded-full bg-grey-shd4 group-hover:bg-grey-shd3 group-focus:border group-focus:border-solid group-focus:border-primary-shd1"
  >
    <span
      className="absolute left-1 rounded-full transition duration-300 w-4 h-4 bottom-1 bg-white"
    />
  </span>
</label>
`;
